packages/utils/node-resolver-core/src/NodeResolver.js

```diff
@@ -598,8 +611,7 @@ export default class NodeResolver {
     // If the package has a `source` field, check if it is behind a symlink.
     // If so, we treat the module as source code rather than a pre-compiled module.
     if (pkg.source) {
-      let realpath = await this.fs.realpath(file);
-      if (realpath === file) {
+      if (!isSymlink) {
         delete pkg.source;
```

Comment:
I'm curious: what fails if you get rid of this logic. I'm wondering if we could get rid of it entirely.

Reply:
What about postponing micro-optimizations for follow up PRs?

Reply:
This test fails: https://github.com/parcel-bundler/parcel/blob/146cffb6896d1f2bcc3f7f4426440b09b61142b7/packages/utils/node-resolver-core/test/resolver.js#L701-L717

Reply:
Someone should sit and read all of these changes to figure out what has changed where that this test does not pass anymore
https://github.com/dishuostec/parcel/compare/resolve-realpath...parcel-bundler:v2